### ISSUE TITLE
Generate Odometry

### ISSUE THREAD

**Issue by jiahao-0204** [ori-drs/planar_mesh#28]
sample points from a plane and do icp odometry to demonstrate.

**Comment by mauricefallon** [comment]
no I mean for your paper to include an experiment "**localising** in a prior point cloud map"

the prior cloud map would be sampled from the planes so you can easily use our icp_odometry package (which supports both odometry and localisation modes)
... other wise you would have to localise directly to the (non-infinite) plane which would be more work.

for example this is localising to a prior map at 0:45:
https://youtu.be/aIeMPeHDUgs?t=43

**Comment by mauricefallon** [comment]
in the far future, you could also do odometry directly using this model (instead of using VILENS) - but this is a far future topic. Not for IROS

**Comment by jiahao-0204** [comment]
> no I mean for your paper to include an experiment "**localising** in a prior point cloud map"
> 
> the prior cloud map would be sampled from the planes so you can easily use our icp_odometry package (which supports both odometry and localisation modes) ... other wise you would have to localise directly to the (non-infinite) plane which would be more work.
> 
> for example this is localising to a prior map at 0:45: https://youtu.be/aIeMPeHDUgs?t=43

I see. so I 
- run through a sequence using my algorithm
- sample points from my planar reconstruction to generate a point cloud reconstruction
- localise a random scan against the generated point cloud to test my reconstruction

is that correct?

**Comment by mauricefallon** [comment]
yes, but instead of "localise _a random scan_ against the generated point cloud to test my reconstruction", you can localise a full recording to the point cloud reconstruction.

For example in this video the blue cloud is fixed and the purple scans are continously localised to it:
https://www.youtube.com/watch?v=SuH2jc4yZ4Y
This is the mode we use to localise Spot in a prior map.

**Comment by jiahao-0204** [comment]
save for future
